meta/saisanitycheck.c

```diff
@@ -849,14 +849,13 @@ void check_attr_conditions(
             /* condition needs to be changed to validonly since flags should be CREATE_AND_SET */
 
             if (md->objecttype != SAI_OBJECT_TYPE_MIRROR_SESSION &&
-                    md->objecttype != SAI_OBJECT_TYPE_HOSTIF_TRAP &&
-                    md->objecttype != SAI_OBJECT_TYPE_HOSTIF_USER_DEFINED_TRAP)
+                    md->objecttype != SAI_OBJECT_TYPE_HOSTIF_TRAP)
             {
                 META_ASSERT_FAIL(md, "marked as conditional on non mirror session /hostif_trap/hostif_user_defined_trap, but invalid creation flags: 0x%u", md->flags);
             }
 
             break;
-
+        case SAI_ATTR_FLAGS_MANDATORY_ON_CREATE | SAI_ATTR_FLAGS_CREATE_ONLY | SAI_ATTR_FLAGS_KEY:
```

Comment:
Here is the motivation
Trap object can either be regular trap or user defined trap (controlled by SAI_HOSTIF_TRAP_ATTR_IS_USER_DEFINED_TRAP)
When it is regular trap, SAI_HOSTIF_TRAP_ATTR_TRAP_TYPE field is a key, meaning there can be exactly one instance of each type.
When it is user defined trap, we allow creating multiple instances of each SAI_HOSTIF_TRAP_ATTR_USER_DEFINED_TYPE.
So the key field is conditional if it's trap or user defined trap
Please explain whether I should remove the key field, but then we won't be able to specify we allow just one of each type